### DISCUSSION TITLE
How can I display a human-readable error?

### DISCUSSION THREAD

**Discussion by nandorojo** [urql-graphql/urql#2091] [Q&A] [answered]
I'm not sure how to read the `error?.message` to display to users.

It seems that in no part of the errors object is there a way to parse the `message` returned from the graphql server.

I'm currently doing this (horrible lol) string parse.

```ts
  const errorMessage = (() => {
    let message = result.error?.message
    if (message) {
      if (message.startsWith('[GraphQL] {')) {
        try {
          const json = JSON.parse(
            // replace all instances of ' with "
            message.replace('[GraphQL] {', '{').replace(/'/g, `"`)
          )
          message = json.message
        } catch (e) {
        }
      }
    }
    return message
  })()
```

Is there a better way to do this? I don't see it in the docs.

**Comment by kitten** [comment, marked as answer]
That's documented in these two places ✌️
- https://formidable.com/open-source/urql/docs/basics/errors/
- https://formidable.com/open-source/urql/docs/api/core/#combinederror

The message is purely for debugging, but it's likely the `graphQLErrors` array property you're looking for, which will give you the fully parsed list of `GraphQLError`s that your API sent 

On a side note, it's not a great idea to parse messages arbitrarily, since they may be any string message. The spec actually has something to say about error properties: https://spec.graphql.org/October2021/#sec-Errors

Specifically, it suggests adding a custom property to your `GraphQLError`'s `extensions` record, which will then be received as JSON, as in, it'll simply be passed through

**Reply by nandorojo** [reply]
I did see those links, however I didn't see anything about providing a human-readable error to users.

When I log the error returned from urql, I see this:

![image](https://user-images.githubusercontent.com/13172299/142709025-d272b6b5-f6fe-4c45-85d4-45091ae30842.png)

From there, I couldn't find a suitable way to get an error to display to the user.

> On a side note, it's not a great idea to parse messages arbitrarily, since they may be any string message. The spec actually has something to say about error properties: https://spec.graphql.org/October2021/#sec-Errors

Definitely agree, just wasn't sure what else to do.

> Specifically, it suggests adding a custom property to your GraphQLError's extensions record, which will then be received as JSON, as in, it'll simply be passed through

This is good to know. I'll forward that detail to someone who knows the backend side. So the extension can just be any JSON that gets passed on, correct?
